golos.publication/objects.hpp

```diff
@@ -114,9 +114,15 @@ struct rewardrules {
 struct forumprops_record {
     forumprops_record() = default;
 
+    forumprops_record(const forumprops& props) {
+        contract_for_reputation = props.contract_for_reputation;
+        contract_for_blocking = props.contract_for_blocking;
+    }
+
     name contract_for_reputation = name();
+    name contract_for_blocking = name();
```

Comment:
I propose not to create unnecessary entities and use one smart contract for reputation and blocking. All the same, when the method is implemented as parameters, it will be one field. For now let's just rename `contract_for_reputation` into `social_contract`.

Reply:
@s-medvedev I was already proposing `social_contract` in past.
However, 2 fields are more flexible...